connector-runtime/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorManager.java

```diff
@@ -0,0 +1,125 @@
+package io.camunda.connector.runtime.inbound.lifecycle;
+
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.impl.inbound.InboundConnectorProperties;
+import io.camunda.connector.runtime.inbound.importer.ProcessDefinitionInspector;
+import io.camunda.connector.runtime.util.inbound.InboundConnectorFactory;
+import io.camunda.operate.dto.ProcessDefinition;
+import io.camunda.operate.exception.OperateException;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class InboundConnectorManager {
+  private static final Logger LOG = LoggerFactory.getLogger(InboundConnectorManager.class);
+
+  private final InboundConnectorFactory connectorFactory;
+  private final InboundConnectorContext connectorContext;
+  private final ProcessDefinitionInspector processDefinitionInspector;
+
+  // TODO: consider using external storage instead of these collections to allow multi-instance setup
+  private final Set<Long> registeredProcessDefinitionKeys = new HashSet<>();
+  private final Map<InboundConnectorProperties, InboundConnectorExecutable> activeConnectors =
+    new ConcurrentHashMap<>();
+  private final Map<String, Set<InboundConnectorProperties>> activeConnectorsByBpmnId = new HashMap<>();
+
+  public InboundConnectorManager(
+    InboundConnectorFactory connectorFactory,
+    InboundConnectorContext connectorContext,
+    ProcessDefinitionInspector processDefinitionInspector) {
+    this.connectorFactory = connectorFactory;
+    this.connectorContext = connectorContext;
+    this.processDefinitionInspector = processDefinitionInspector;
+  }
+
+  /**
+   * Check whether process definition with provided key is already registered
+   */
+  public boolean isProcessDefinitionRegistered(long processDefinitionKey) {
+    return registeredProcessDefinitionKeys.contains(processDefinitionKey);
+  }
+
+  /**
+   * Process a batch of process definitions
+   */
+  public void registerProcessDefinitions(List<ProcessDefinition> processDefinitions) throws OperateException {
```

Comment:
Do we need to check the `processDefinitions` for `null` or if there are no definitions, it will be an empty list?

Reply:
Better safe than sorry 🙂